utils/vaultsigner/vaultsigner.go

```diff
@@ -18,65 +12,14 @@ import (
 	"golang.org/x/crypto/ed25519"
 )
 
-// VaultSigner an ed25519 signer / verifier that uses the vault transit backend
-type VaultSigner struct {
-	Client     *api.Client
-	KeyName    string
-	KeyVersion uint
-}
-
-// Sign the included message using the vault held keypair. rand and opts are not used
-func (vs *VaultSigner) Sign(rand io.Reader, message []byte, opts crypto.SignerOpts) ([]byte, error) {
-	response, err := vs.Client.Logical().Write("transit/sign/"+vs.KeyName, map[string]interface{}{
-		"input": base64.StdEncoding.EncodeToString(message),
-	})
-	if err != nil {
-		return []byte{}, err
-	}
-
-	sig := response.Data["signature"].(string)
-
-	return base64.StdEncoding.DecodeString(strings.Split(sig, ":")[2])
-}
-
-// Verify the included signature over message using the vault held keypair. opts are not used
-func (vs *VaultSigner) Verify(message, signature []byte, opts crypto.SignerOpts) (bool, error) {
-	response, err := vs.Client.Logical().Write("transit/verify/"+vs.KeyName, map[string]interface{}{
-		"input":     base64.StdEncoding.EncodeToString(message),
-		"signature": fmt.Sprintf("vault:v%d:%s", vs.KeyVersion, base64.StdEncoding.EncodeToString(signature)),
-	})
-	if err != nil {
-		return false, err
-	}
-
-	return response.Data["valid"].(bool), nil
-}
-
-// String returns the public key as a hex encoded string
-func (vs *VaultSigner) String() string {
-	return hex.EncodeToString(vs.Public().(ed25519.PublicKey))
-}
-
-// Public returns the public key
-func (vs *VaultSigner) Public() crypto.PublicKey {
-	response, err := vs.Client.Logical().Read("transit/keys/" + vs.KeyName)
-	if err != nil {
-		panic(err)
-	}
-
-	keys := response.Data["keys"].(map[string]interface{})
-	key := keys[strconv.Itoa(int(vs.KeyVersion))].(map[string]interface{})
-	b64PublicKey := key["public_key"].(string)
-	publicKey, err := base64.StdEncoding.DecodeString(b64PublicKey)
-	if err != nil {
-		panic(err)
-	}
-
-	return ed25519.PublicKey(publicKey)
+// WrappedClient holds an api client for interacting with vault
+type WrappedClient struct {
```

Comment:
Why do we have a "WrappedClient" and not just follow the existing model of just passing in a pointer to the api.Client into functions as an argument?  It seems that we are not enriching the state of the api.Client structure.  Wrapping just seems to be adding methods.